### ISSUE TITLE
Test/Validation Loss doesn't converge?

### ISSUE THREAD

**Issue by croros** [Hippogriff/3DCSGNet#3]
Hello,

I tried training 3DCSGNet with the default parameters. The training loss appears to converge to zero and overall decreases with each epoch, but the validation loss doesn't decrease and seems to flatline around a some positive value (in my case 16 or 17). Is this normal? Are there any parameters from the default yml that I should change to improve this?

Thanks

**Comment by visHUSTLE** [comment]
@croros I think I am facing a similar problem. Were you able to resolve it?
@Hippogriff I am Testing your pre-trained model and i get the IOU score as 0 and pred.txt  file as empty

**Comment by Hippogriff** [comment]
I hope you have installed the requirements. Furthermore, the dimension (parameter) of the primitives are only limited to the keys provided in "data/primitives.h5" file. No other parameters will give you correct output, because the primitives are cached in primitives.h5.

**Comment by visHUSTLE** [comment]
@Hippogriff How can Test this model on Unknown dimensions. Isn't the voxels are being created on the go for Unknown dimensions? 
Also What does exactly does the primitives.h5 consists of? Does it have corresponding labels for each voxelized representation?

**Comment by Hippogriff** [comment]
> @Hippogriff How can Test this model on Unknown dimensions. Isn't the voxels are being created on the go for Unknown dimensions?
No. Creating voxels on the go was a slow process. So, I had to cache primitives at gridded location and scales in the primitives.h5 file.

Also What does exactly does the primitives.h5 consists of? Does it have corresponding labels for each voxelized representation?
Yes.

**Comment by visHUSTLE** [comment]
@Hippogriff Thanks for the reply. 
Actually, I am planning to train the data on different labels
I am gonna replace all the labels that were created based of sketchup representation with a new one 

For eg: The equivalent of 
sp(40,40,24,20)  is  SphereBody.Create(Point.Create(MM(40),MM(40),MM(24)), 20)

Do you think, that can be done?

**Comment by Hippogriff** [comment]
I think so. Note that the idea behind discretizing the space is to control the output space of the parser and have only one categorical cross entropy loss.
As long as you have reasonably small output space, things should work fine.
